mlrun/model_monitoring/db/tsdb/timescaledb/timescaledb_schema.py

```diff
@@ -302,9 +307,10 @@ def _get_records_query(
                 if interval and agg_funcs and use_pre_aggregates:
                     # For pre-aggregates, use column names as-is since they should already be
                     # the correct names from the continuous aggregate view
+                    source_time_col = timestamp_column or self.time_column
                     modified_columns = []
                     for col in columns_to_filter:
-                        if col == self.time_column:
+                        if col == source_time_col:
```

Comment:
I think you don't need that `source_time_col` at all and you can just use the `time_col`
```suggestion
                        if col == time_col:
```

Reply:
Fixed